tests/management/principal/test_it_service.py

```diff
@@ -0,0 +1,334 @@
+        for rci in request_client_ids:
+            request_client_ids_str.add(str(rci))
+
+        # Assert that all the service accounts were flagged as not present in the group.
+        for key, value in result.items():
+            # Make sure the specified client IDs are in the set.
+            self.assertEqual(
+                True,
+                key in request_client_ids_str,
+                "expected to find the specified client ID from the request in the returning result",
+            )
+            # Make sure they are all set to "false" since there shouldn't be any of those client IDs in the group.
+            self.assertEqual(
+                False,
+                value,
+                "the client ID should have not been found in the group, since the group had no service accounts",
+            )
+
+    def test_generate_service_accounts_report_in_group_mixed_results(self):
+        """Test that the function under test is able to correctly flag the sevice accounts when there are mixed results"""
+        principal_1 = Principal.objects.create(username="user-1", tenant=self.tenant)
+        principal_2 = Principal.objects.create(username="user-2", tenant=self.tenant)
+        principal_3 = Principal.objects.create(username="user-3", tenant=self.tenant)
+
+        client_uuid_1 = uuid.uuid4()
+        client_uuid_2 = uuid.uuid4()
+        sa_1 = Principal.objects.create(
+            username=f"service-account-{client_uuid_1}",
+            service_account_id=client_uuid_1,
+            type="service-account",
+            tenant=self.tenant,
+        )
+        sa_2 = Principal.objects.create(
+            username=f"service-account-{client_uuid_2}",
+            service_account_id=client_uuid_2,
+            type="service-account",
+            tenant=self.tenant,
+        )
+        # Create more service accounts that should not show in the results, since they're not going to be specified in
+        # the "client_ids" parameter.
+        sa_3 = Principal.objects.create(
+            username=f"service-account-{uuid.uuid4()}",
+            service_account_id=uuid.uuid4(),
+            type="service-account",
+            tenant=self.tenant,
+        )
+        sa_4 = Principal.objects.create(
+            username=f"service-account-{uuid.uuid4()}",
+            service_account_id=uuid.uuid4(),
+            type="service-account",
+            tenant=self.tenant,
+        )
+        sa_5 = Principal.objects.create(
+            username=f"service-account-{uuid.uuid4()}",
+            service_account_id=uuid.uuid4(),
+            type="service-account",
+            tenant=self.tenant,
+        )
+
+        # Create a set with the service accounts that will go in the group. It will make it easier to make assertions
+        # below.
+        group_service_accounts_set = {str(sa_1.service_account_id), str(sa_2.service_account_id)}
+
+        # For this test, we don't associate any service accounts to it.
```

Comment:
not sure if I understand this comment? below some SA are added into group

Reply:
Refactored!